github/MainClass.py

```diff
@@ -808,6 +808,7 @@ def get_app(self, slug=github.GithubObject.NotSet):
         if slug is github.GithubObject.NotSet:
             # with no slug given, calling /app returns the authenticated app,
             # including the actual /apps/{slug}
+            assert self.__requester._Requester__auth.token_type == 'Bearer', "Only AppAuth authentication support get_app without a slug."
             headers, data = self.__requester.requestJsonAndCheck("GET", "/app")
             return GithubApp.GithubApp(self.__requester, headers, data, completed=True)
```

Comment:
Let's not break user code that use this method without providing `slug`(it should have worked with `Github(jwt=GithubIntegration(...).create_jwt()`, which is deprecated).

Let's keep slug optional, but raise a deprecated warning:
```suggestion
            warnings.warn(
                "Argument slug is mandatory, calling this method without the slug argument is deprecated, please use "
                "github.GithubIntegration(auth=github.Auth.AppAuth(...)).get_app() instead",
                category=DeprecationWarning,
            )
            return GithubIntegration(auth=self.auth).get_app()
```